mllib/src/test/scala/org/apache/spark/ml/feature/BitSamplingSuite.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.feature
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.ml.linalg.{Vector, Vectors}
+import org.apache.spark.mllib.util.MLlibTestSparkContext
+
+class BitSamplingSuite extends SparkFunSuite with MLlibTestSparkContext {
+  test("BitSampling") {
+    val data = {
+      for (i <- 0 to 10) yield Vectors.sparse(10, (0 until i).map((_, 1.0)))
+    }
+    val df = spark.createDataFrame(data.map(Tuple1.apply)).toDF("keys")
+
+    val bs = new BitSampling()
+      .setSampleSize(3)
+      .setInputCol("keys")
+      .setOutputCol("values")
+      .setSeed(0)
```

Comment:
It'd be better not to use seed 0.  I think it's probably fine here, but I have seen RNGs in Java which fail silently when given seed 0.